### ISSUE TITLE
How to render the tiles?

### ISSUE THREAD

**Issue by distancerunner** [Overv/openstreetmap-tile-server#12]
I made the "Setting up the server" command and all was running for more than 500seconds and it was finishing successful.

Now I'm only getting empty tiles (I only used the data from Georgia)
http://188.68.46.145/tile/12/2563/1518.png -> empty tile
http://188.68.46.145/tile/0/0/0.png -> the world tile (working)

I downloaded the .osm.pbf file from georgia and run:
docker run -v /absolute/path/to/downloads/georgia.osm.pbf:/data.osm.pbf ...

But there was a strange behavior: it started to download some GB of data. For me it looked like the container didn't find the downloaded file. Did I something wrong?

**Comment by Overv** [comment]
Did it state the following message as part of the download process?

>WARNING: No import file at /data.osm.pbf, so importing Luxembourg as example..

**Comment by distancerunner** [comment]
I run stat stat command:

Datei: /home/someuser/Downloads/georgia-latest.osm.pbf
  Größe: 177550561 	Blöcke: 346792     EA Block: 4096   Normale Datei
Gerät: 803h/2051d	Inode: 436113      Verknüpfungen: 1
Zugriff: (0664/-rw-rw-r--)  Uid: ( 1000/  someuser)   Gid: ( 1000/  someuser)
Zugriff    : 2019-01-07 15:34:35.068393335 +0100
Modifiziert: 2019-01-07 15:33:22.700483799 +0100
Geändert   : 2019-01-07 15:33:22.800483698 +0100
 Geburt    : -

**Comment by Overv** [comment]
The output looks fine to be and there is no warning about `/data.osm.pbf` not existing. Are you sure that the data for Georgia is not available in a tile like 8/159/95.png?

**Comment by distancerunner** [comment]
Since yesterday, the new container is still running:
http://188.68.46.145/tile/8/159/95.png -> empty tile
http://188.68.46.145/tile/12/2563/1518.png -> empty tile
http://188.68.46.145/tile/0/0/0.png -> the world tile (working)

Where can I find the files in Filesystem? 
My server is using Ubuntu 18.04.1 LTS
Can I re-render tiles with a command?

Thanks for your help. I'm writing a App for a Georgian NGO, so actually I using a expensive provider and we need to find a cheaper solution.

**Comment by Overv** [comment]
>Where can I find the files in Filesystem?

I'm not sure if there is a way to directly access Docker volumes from the local file system. If you want to be able to easily access the imported data, then I recommend mounting a folder instead of a volume like `openstreetmap-data`.

>Can I re-render tiles with a command?

Already rendered tiles are stored in the cache, which can be cleared by simply restarting the container or clearing the folder `/var/lib/mod_tile` inside of it.

I'm not sure what is going on in your case that causes the data not to appear. Can you check if the Luxembourg sample data *does* show up on the map if you use that instead of Georgia?

**Comment by distancerunner** [comment]
I started a new try. I installed CentOS7 and used this Tutorial:
https://www.openstreetmap.org/user/SomeoneElse/diary/45070
My Vserver is a:
1 Core
2GB RAM
20GB HDD

I got this error, and I think, this error occured also in Ubuntu at my very first try. Maybe it is a issue with an empty Database:
Using lua based tag processing pipeline with script /home/renderer/src/openstreetmap-carto/openstreetmap-carto.lua
Using projection SRS 3857 (Spherical Mercator)
Setting up table: planet_osm_point
Setting up table: planet_osm_line
Setting up table: planet_osm_polygon
Setting up table: planet_osm_roads
Allocating memory for dense node cache
Allocating dense node cache in one big chunk
Out of memory for dense node cache, reduce --cache size
Error occurred, cleaning up

**Comment by Overv** [comment]
Right, yeah, you will need more memory than that.

**Comment by distancerunner** [comment]
Now I run all commands on a local PC (32GB RAM and i7 Core)
 I also used Luxemburg as source.

I got the same result, only white tiles are generated for Georgia and for Luxemburg.

**Comment by Overv** [comment]
In that case I don't know what the issue is.

**Comment by Istador** [comment]
I think I might have figured it out.

---

First of all your posted output contained these lines:
```
createuser: creation of new role failed: ERROR: role "renderer" already exists
createdb: database creation failed: ERROR: database "gis" already exists
ERROR: extension "postgis" already exists
ERROR: extension "hstore" already exists
```

This indicates that you haven't deleted the named docker volume `openstreetmap-data` from your previous run(s).

```
$ docker volume rm openstreetmap-data
Error response from daemon: remove openstreetmap-data: volume is in use - [a44468f0970cc9cc8355aa2f631d58433f2f084a1fe32a3cfdd8aa7e2a619af5, 3bdeb933e5d53be52c52b5050c19842b55b93fb92c6099bdfb26a65d24bbca5e]
$ docker rm a44468f0970cc9cc8355aa2f631d58433f2f084a1fe32a3cfdd8aa7e2a619af5
a44468f0970cc9cc8355aa2f631d58433f2f084a1fe32a3cfdd8aa7e2a619af5
$ docker rm 3bdeb933e5d53be52c52b5050c19842b55b93fb92c6099bdfb26a65d24bbca5e
3bdeb933e5d53be52c52b5050c19842b55b93fb92c6099bdfb26a65d24bbca5e
$ docker volume rm openstreetmap-data
openstreetmap-data
```

---

Secondly, I believe, that you used the PBF file of Georgia (USA) but used tile coordinates from Georgia (Europe).

According to the log you posted it took you 523 seconds to import:
```
Osm2pgsql took 523s overall
```

When I import Georgia (USA) it takes 428 seconds to import (169 MiB PBF):
```
Osm2pgsql took 428s overall
```
and I have the same result as you, meaning empty tiles at `8/159/95` and `12/2563/1518`

Can you try to fetch the tile `6/17/25` to see if Georgia (USA) is non-empty.

---

When I import Georgia (Europe) it takes 96 seconds to import (45 MiB PBF):
```
Osm2pgsql took 96s overall
```
and both tiles are non-empty.

Successful output:

```
$ d="openstreetmap-data"
$ u="https://download.geofabrik.de/europe/georgia-latest.osm.pbf"
$ f="georgia-latest.osm.pbf"
$ pgdb="/var/lib/postgresql/10/main"
$ img="overv/openstreetmap-tile-server"
$ docker version
[...]
Server: Docker Engine - Community
 Engine:
  Version:          18.09.1
  API version:      1.39 (minimum version 1.12)
  Go version:       go1.10.6
  Git commit:       4c52b90
  Built:            Wed Jan  9 19:02:44 2019
  OS/Arch:          linux/amd64
  Experimental:     false
$ docker volume create "$d"
openstreetmap-data
$ wget "$u"
[...]
2019-01-24 22:41:45 (10,3 MB/s) - ‘georgia-latest.osm.pbf’ saved [46871899/46871899]
$ docker run -v "/`pwd`/$f":/data.osm.pbf -v "$d":"$pgdb" "$img" import
Unable to find image 'overv/openstreetmap-tile-server:latest' locally
latest: Pulling from overv/openstreetmap-tile-server
[...]
Digest: sha256:4df764e587e3c819368da8a0607d7a2293dd22bf9ba9d656e80caceb9231b66b
Status: Downloaded newer image for overv/openstreetmap-tile-server:latest
 * Starting PostgreSQL 10 database server
   ...done.
CREATE EXTENSION
CREATE EXTENSION
ALTER TABLE
ALTER TABLE
osm2pgsql version 0.96.0 (64 bit id space)
[...]
Reading in file: /data.osm.pbf
Using PBF parser.
[...]
Osm2pgsql took 96s overall
$ docker run -p 8080:80 -v "$d":"$pgdb" -d "$img" run
3bdeb933e5d53be52c52b5050c19842b55b93fb92c6099bdfb26a65d24bbca5e
$ wget http://localhost:8080/tile/0/0/0.png
[...]
$ wget http://localhost:8080/tile/8/159/95.png
[...]
$ wget http://localhost:8080/tile/12/2563/1518.png
[...]
$ ls -lan .
-rw-r--r-- 1 1000 1000     8908 Jan 24 22:56 0.png
-rw-r--r-- 1 1000 1000    14067 Jan 24 22:56 1518.png
-rw-r--r-- 1 1000 1000    21386 Jan 24 22:56 95.png
-rw-r--r-- 1 1000 1000 46871899 Jan 24 01:38 georgia-latest.osm.pbf
$ sha1sum *
47eabf10bc79dc553eaefc18d6af009f6804b375  0.png
3874bd07df609263ee873a0de7c69700e88607bc  1518.png
d63f45299787e9121e02ee394b0cf4e4d7781632  95.png
7705662756eff6244c44c4f1239ec128a6e0b593  georgia-latest.osm.pbf
```

**Comment by distancerunner** [comment]
Thanks for your further investigations. I was not able to understand the docker structure fast enough. So maybe I missed something or deleted it not very well on the second run.
I ended up by using a docker container from https://openmaptiles.org/ . I generated my own tiles/mbtiles files from Georgia(Eastern Europe), Zoom 0-16. It took 10h. Now the tileserver is running well.

Actuall I generating Germany at Zoom 0-16. It is taking actually 12days for 85% and.

But I repeat, thanks for your help!
